### PR TITLE
Regressors 2.0

### LINKED ISSUE #513: Create Changelog
### 🚀 Feature Request

Create Changelog 

### Motivation

Documentation to track new regressors logic

### Proposal

1. Create Changelog including all the improvements in the branch
2. Merge branch into master

### Test cases

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [x] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes
1. TSDataset:
    - Now the list of regressors is stored in the dataset, it can be obtained with `regressors` property. Only the columns from regressors list are regressors
    - New parameter `known_future` in the constructor, indicates regressor columns in df_exog
    - The list of regressors is updated after applying each transform in transform loop using `_update_regressors` method using the following rules:
        - All columns generated by Transform with `FutureMixin` are regressors
        - All columns created from regressors are regressors
        - List of regressors is always the subset of all columns
    - Methods `make_future` and `train_test_split` pass the list of regressors to the generated datasets
    - Methods  `info` and `describe` now also shows the information about regressors
2. Transforms:
    -`FutureMixin` is added to all the transforms the output columns of which are regressors:
        - decomposition: `TrendTransform`
        - encoders: `MeanSegmentEncoderTransform`, `SegmentEncoderTransform`
        - math: `LagTransform`
        - timestamp: `DateFlagsTransform`, `FourierTransform`, `HolidayTransform`, `SpecialDaysTransform`, `TimeFlagsTransform`
     - `BaseFeatureSelectionTransform`, `TreeFeatureSelectionTransform`, `MRMRFeatureSelectionTransform`, `GaleShapleyFeatureSelectionTransform` now has parameter `features_to_use` - there should be listed all the features you want to apply the Transform on.
     - Variables in the feature selection transforms above are renamed replacing the substring "regressor" to "feature"
3. Models:
    - `ProphetModel`, `SARIMAXModel` and all the `Sklearn` models now works only with regressors
    - Adapters(`_SklearnModel`...) now get the list of regressors in method  `fit` and save into `regressor_columns` parameter and use this list in the `forecast`
    - Models pass the list of regressor from the dataset to the Adapter inside the `fit` method  
    - `Catboost` still uses all the columns in the dataset

## Closing issues

closes #513

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,39 +7,41 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
--
--
--
--
--
--
+- Regressors logic to TSDatasets init ([#357](https://github.com/tinkoff-ai/etna/pull/357))
+- `FutureMixin` into some transforms([#361](https://github.com/tinkoff-ai/etna/pull/361))
+- Regressors updating in TSDataset transform loops ([#374](https://github.com/tinkoff-ai/etna/pull/374))
+- Regressors handling in TSDataset `make_future` and `train_test_split`  ([#447](https://github.com/tinkoff-ai/etna/pull/447))
```
Comment (Mr-Geekman) [inline CHANGELOG.md:13]: Too many spaces before link

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,39 +7,41 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ### Added
--
--
--
--
--
--
--
+- Regressors logic to TSDatasets init ([#357](https://github.com/tinkoff-ai/etna/pull/357))
+- `FutureMixin` into some transforms([#361](https://github.com/tinkoff-ai/etna/pull/361))
```
[the remark this answers is not in the document]
Comment (alex-hse-repository) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 1 above)
Comment (alex-hse-repository) [reply CHANGELOG.md]: Fixed

**Review by martins0n — DISMISSED** [review]
👍

**Review by Mr-Geekman — APPROVED** [review]

**Pull request merged**